gap/attr.gi

```diff
@@ -1687,6 +1687,138 @@ function(D)
+          in_core[m] := false;
+        fi;
+        if m < L then
+          m := m + 1;
+        else
+          m := 1;
+        fi;
+      od;
+      n := n + 1;
+    od;
+    cores := ListBlist(cores, in_core);
+    vert_labels := [];
+    for G in cores do
+      vert_labels := Concatenation(vert_labels, DigraphVertexLabels(G));
+    od;
+    return vert_labels;
+  elif IsCompleteDigraph(digraph) then
+    return [1 .. N];
+  elif IsSymmetricDigraph(digraph) and IsBipartiteDigraph(digraph) then
+    return DigraphEdges(digraph)[1];
+  fi;
+
+  if IsDigraphCore(digraph) then
+    return DigraphVertexLabels(digraph);
+  fi;
+
+  lo := function(D)  # lower bound on core size
+    local oddgirth;
+    oddgirth := DigraphOddGirth(D);
+    if oddgirth <> infinity then
+      return oddgirth;
+    fi;
+    return 3;
+  end;
+
+  hom      := [];
+  lo_var := lo(digraph);
+  bottomup := lo_var;
+  N := DigraphNrVertices(digraph);
+  topdown  := N;
+
+  while topdown >= bottomup do
+    HomomorphismDigraphsFinder(digraph,                   # domain copy
+                               digraph,                   # range copy
+                               fail,                      # hook
+                               hom,                       # user_param
+                               1,                         # max_results
+                               bottomup,                  # hint (i.e. rank)
+                               false,                     # injective
+                               DigraphVertices(digraph),  # image
+                               [],                        # partial_map
+                               fail,                      # colors1
+                               fail);                     # colors2
+
+    if Length(hom) = 1 then
+      image    := ImageSetOfTransformation(hom[1], N);
+      return DigraphVertexLabels(digraph){image};
+    fi;
+
+    HomomorphismDigraphsFinder(digraph,                   # domain copy
+                               digraph,                   # range copy
+                               fail,                      # hook
+                               hom,                       # user_param
+                               1,                         # max_results
+                               topdown,                   # hint (i.e. rank)
+                               false,                     # injective
+                               DigraphVertices(digraph),  # image
+                               [],                        # partial_map
+                               fail,                      # colors1
+                               fail);                     # colors2
+    if Length(hom) = 1 then
+      image    := ImageSetOfTransformation(hom[1], N);
+      digraph  := InducedSubdigraph(digraph, image);
+      N        := DigraphNrVertices(digraph);
+      Unbind(hom[1]);
+      lo_var := lo(digraph);
+    fi;
+
+    if topdown - 1 < N then
```

Comment:
Here (I think?) you can just do:
```
topdown := Minimum(topdown - 1, N);
bottom := Maximum(bottomup + 1, lo_var);
```

Reply:
I agree that using the minimum and maximum functions is more readable, but I remember @james-d-mitchell suggested that sometimes they can be less efficient than using conditionals?

Reply:
Possibly, but I think any difference in speed would be negligible, whereas I think the `Minimum` etc significantly benefits readability and future maintenance.